src/server/streamableHttp.ts

```diff
@@ -794,13 +803,21 @@ export class StreamableHTTPServerTransport implements Transport {
      * Close an SSE stream for a specific request, triggering client reconnection.
      * Use this to implement polling behavior during long-running operations -
      * client will reconnect after the retry interval specified in the priming event.
+     *
+     * @param requestId - The request ID whose stream should be closed
+     * @param retryInterval - Optional retry interval in milliseconds to send before closing.
+     *                        If provided, sends a retry field to override the transport default.
      */
-    closeSSEStream(requestId: RequestId): void {
+    closeSSEStream(requestId: RequestId, retryInterval?: number): void {
         const streamId = this._requestToStreamMapping.get(requestId);
         if (!streamId) return;
 
         const stream = this._streamMapping.get(streamId);
         if (stream) {
+            // If a custom retry interval is provided, send it before closing
+            if (retryInterval !== undefined) {
```

Comment:
Doesn't this cause problems for the client?
If the "event" field is missing, the SSE spec says that it should be treated as a "message" event.